### PR TITLE
added-sector-service

### PR DESCRIPTION
**Pull request by udisinghania**

### CODE REVIEW

File: backend/sector-service/src/main/java/com/example/sectorservice/controllers/SectorController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/sector")
+public class SectorController {
+
+    @Autowired
+    private SectorService sectorService;
+
```
Comment (ujjwal0507) [inline backend/sector-service/src/main/java/com/example/sectorservice/controllers/SectorController.java:16]: Controller routes shoudl return Response Entity

File: backend/sector-service/src/test/java/com/example/sectorservice/SectorServiceApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.example.sectorservice;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
```
Comment (ujjwal0507) [inline backend/sector-service/src/test/java/com/example/sectorservice/SectorServiceApplicationTests.java:5]: Delete the empty class.

File: backend/sector-service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
```
Comment (ujjwal0507) [inline backend/sector-service/pom.xml:6]: Chagne the parent so that it refers to the parent pom file.

File: backend/sector-service/pom.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-jdbc</artifactId>
+		</dependency>
+		<dependency>
```
Comment (ujjwal0507) [inline backend/sector-service/pom.xml:28]: Dependency alredy present in the parent pom.

File: backend/sector-service/pom.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>mysql</groupId>
+			<artifactId>mysql-connector-java</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
```
Comment (ujjwal0507) [inline backend/sector-service/pom.xml:42]: Dependency already present in the parent pom

File: backend/sector-service/pom.xml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
```
Comment (ujjwal0507) [inline backend/sector-service/pom.xml:49]: Build plugins are present in the parent pom file.

**Review by ujjwal0507 — CHANGES_REQUESTED** [review]
Please see the comments and do the required changes.

File: backend/sector-service/src/main/java/com/example/sectorservice/entities/Sector.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        return brief;
+    }
+
+    public void setBrief(String brief) {
+        this.brief = brief;
+    }
+
+    @Override
+    public String toString() {
+        return "Sector{" +
```
Comment (ibmpariwesh) [inline backend/sector-service/src/main/java/com/example/sectorservice/entities/Sector.java:59]: use Stringbuilder

File: backend/sector-service/src/main/java/com/example/sectorservice/services/SectorService.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        List<Sector> sectors = new ArrayList<Sector>();
+        sectorRepository.findAll().forEach(sector1 -> sectors.add(sector1));
+        return sectors;
+    }
+
+    public Sector getSectorById(String id)
+    {
+        Optional<Sector> sector = sectorRepository.findById(id);
+        if(sector.isPresent())
```
Comment (ibmpariwesh) [inline backend/sector-service/src/main/java/com/example/sectorservice/services/SectorService.java:30]: use ternary operator

**Review by ujjwal0507 — APPROVED** [review]
Okay

**Pull request merged**